### DISCUSSION TITLE
White screen splashes when using dark mode

### DISCUSSION THREAD

**Discussion by megastallman** [Eloston/ungoogled-chromium#1355] [Q&A] [answered]
Hi guys! That is basically a Chromium issue, that was half-fixed and broken again by its devs.
I'm using "chrome://flags/ -> Force Dark Mode for Web Contents -> Selective Inversion of Everything" setting, but earlier - was using the Dark Reader plugin instead. The problem is the same, when I open a new page - I see the white canvas only. It can last up to 20 seconds if the site is really heavy and the connection is too slow. Or doesn't happen at all if the site is precached already or is lightweight enough. So at some point white color is hardcoded to be used by default. And it doesn't respect the Dark Mode option.
Does anyone have an idea, at least which component is that? I'm comfortable with UC rebuild process, so I can experiment with its code.

P.S. - Thanks again for the best browser in the universe.

**Comment by kramred** [comment]
You could first test one of the dark mode new tab extensions, like this one: https://github.com/dandydanny/darknewtab – it uses the mechanism described [here (Overriding Chrome pages)](https://developer.chrome.com/docs/extensions/mv2/override/)

Or, if that is not used for opening a link in a new window/tab (probably isn't) – try looking in one of these directories of the unpacked sources: `chrome/browser/resources/ntp4` or `chrome/browser/resources/new_tab_page` for something that could be used as default empty page.

Although, the background for opening a link in an empty new window/tab might be defined somewhere else in `chrome/browser` or even other places… maybe in `chrome/browser/ui/webui/new_tab_page/new_tab_page_ui.cc` in `NewTabPageUI::UpdateBackgroundColor` (?)

**Comment by megastallman** [comment]
Thanks for your time Kramred, although looks like the white flash happens much earlier than this code gets run.
So, my build script pre-deletes the `build/src/out/Default` directory, to avoid any leftovers.
Darknewtab extension - just makes the tab dark only after that white flash. As for `chrome/browser/resources/ntp4` and `chrome/browser/resources/new_tab_page` - I've replaced `background-color: white` with `green`, but still have never seen that color. Also - commented out(tried differet combinations) the `UpdateBackgroundColor(theme);` line in new_tab_page_ui.cc. But still - nothing changes. As it was before patch, I'm still getting:
1. White canvas - 1-10 seconds.
2. Dark canvas - less than a second(probably the Google's partial fix, that I haven't noticed last year).
3. Page contents get loaded.

**Comment by BachoSeven** [comment]
Relevant (old) [patch](https://github.com/hbtlabs/chromium-white-flash-fix), which I link because it contains references to (old) issues and threads on this (old) problem.
A newer issue can be found [here](https://bugs.chromium.org/p/chromium/issues/detail?id=853986).

**Reply by BachoSeven** [reply]
possibly relevant: https://chromium-review.googlesource.com/c/chromium/src/+/1418650

**Comment by kramred** [comment]
Okay, I think I found what controls this → [Themes](https://developer.chrome.com/docs/extensions/mv3/themes/)

The minimal way to change only the background color of new windows/tabs to `rgb(0, 0, 0)` would be creating a `manifest.json` file with the content below in an empty folder and loading it via `chrome://extensions/` "Load unpacked"

```json
{
  "manifest_version": 2,
  "version": "3",
  "name": "ntp_background_000",
  "theme": {
    "colors": {
      "ntp_background": [0, 0, 0]
    }
  }
}
```

See the article [What are themes?](https://developer.chrome.com/docs/extensions/mv3/themes/) and e.g. the source code of the [Just Black](https://chrome.google.com/webstore/detail/just-black/aghfnjkcakhmadgdomlmlhhaocbkloab?hl=en) Chrome Theme with the following `manifest.json` for a more fitting overall design idea:

```json
{
"update_url": "https://clients2.google.com/service/update2/crx",

  "manifest_version": 2,
  "version": "3",
  "name": "Just Black",
  "description": "Midnight monochrome",

 "theme": {
    "colors": {
      "frame": [ 0,0,0],
      "toolbar": [45,45,45],
      "ntp_background": [0,0,0],


      "tab_text": [255,255,255],
      "bookmark_text": [255,255,255],
      "tab_background_text": [255,255,255],
      "tab_background_text_inactive": [ 255,255,255],
      "tab_background_text_incognito": [ 255,255,255 ],
      "tab_background_text_incognito_inactive": [ 255,255,255],


      "ntp_text": [125,134,142]

       },      

    "tints" : {
      "buttons": [ 0.2, 0.2, 0.9 ],
      "frame_incognito": [ -1, -1, -1 ],
      "frame_inactive": [ -1, -1, -1],
      "frame_incognito_inactive": [  -1, -1, -1]
    
     }
  }
}
```

You can revert the Theme in `chrome://settings/appearance` under "Theme"

The Chromium source file [chrome/browser/themes/browser_theme_pack.cc](https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/themes/browser_theme_pack.cc;l=220) specifies the "Strings used by themes to identify colors in the JSON."

**Comment by megastallman** [comment, marked as answer]
Probably something's fixed on version 95, portable Linux version.
Running UC as:
```
/somewhere/chrome --enable-features=OverlayScrollbar --force-dark-mode  %U
```
At least now, I can resize the UC window on youtube and other sites, wait for long DNS responces and static contents load, and finally there are no white splashes. The canvas begins drawing black, according to my KDE theme. Thanks everyone!
